typescript/trans/typechecking/typescript.nabl2

```diff
@@ -253,7 +255,36 @@ rules
       Return{"function"} <- function_scope,
       Return{"function"} : returnTy,
       Map1 [[ body ^ (function_scope) ]].
-     
+    
+    
+    [[ ClassDeclaration(None(), _, parent, members) ^ (s) ]] :=
+      new clazz,
+      clazz -P-> s,
+      Map1 [[ members ^ (clazz) ]],
+      ClassParent [[ parent ^ (s): parentTy ]].
+      
+    [[ ClassDeclaration(Some(name), _, parent, members) ^ (s) ]] :=
+      new clazz,
+      clazz -P-> s,
+      Value{name} <- s,
+      classTy == RECORD(clazz),
+      Value{name} : classTy !,
+      Type{name} <- s,
```

Comment:
Just checked, you can assign a record to a class if the fields match, but not the other way around.